verl/experimental/agent_loop/tool_agent_loop.py

```diff
@@ -118,91 +139,119 @@ async def run(self, sampling_params: dict[str, Any], **kwargs) -> AgentLoopOutpu
-
-                # Handle video data
-                if tool_response.video:
-                    # Currently not supported, raise informative error
-                    logger.warning("Multimedia type 'video' is not currently supported. Only 'image' is supported.")
-                    raise NotImplementedError(
-                        "Multimedia type 'video' is not currently supported. Only 'image' is supported."
+                    response_ids = await self.loop.run_in_executor(
+                        None,
+                        lambda messages=tool_messages: self.tokenizer.apply_chat_template(
+                            messages, add_generation_prompt=True, tokenize=True
+                        ),
                     )
+                response_ids = response_ids[len(self.system_prompt) :]
 
-            # append tool_response_ids
-            if self.processor is not None:
-                raw_tool_response = await self.loop.run_in_executor(
-                    None,
-                    lambda messages=tool_messages: self.processor.apply_chat_template(
-                        messages, add_generation_prompt=True, tokenize=False, **self.apply_chat_template_kwargs
-                    ),
-                )
-                # Use only the new images from this turn for processing tool responses
-                current_images = new_images_this_turn if new_images_this_turn else None
-                model_inputs = self.processor(text=[raw_tool_response], images=current_images, return_tensors="pt")
-                tool_response_ids = model_inputs.pop("input_ids").squeeze(0).tolist()
-            else:
-                tool_response_ids = await self.loop.run_in_executor(
-                    None,
-                    lambda messages=tool_messages: self.tokenizer.apply_chat_template(
-                        messages, add_generation_prompt=True, tokenize=True, **self.apply_chat_template_kwargs
-                    ),
-                )
-            tool_response_ids = tool_response_ids[len(self.system_prompt) :]
-
-            # NOTE: last turn should not be user turn, or the EOS token reward
-            # can't be propagated to previous token in GAE.
-            if len(response_mask) + len(tool_response_ids) >= self.response_length:
-                break
+                # NOTE: last turn should not be user turn, or the EOS token reward
+                # can't be propagated to previous token in GAE.
+                if len(response_mask) + len(response_ids) >= self.response_length:
+                    break
 
-            prompt_ids += tool_response_ids
-            response_mask += [0] * len(tool_response_ids)
-            user_turns += 1
+                prompt_ids += response_ids
+                response_mask += [0] * len(response_ids)
+                # user_turns += 1
+                if self.interaction_config_file and not tool_calls and should_terminate_sequence:
```

Comment:
This code block is under `if tool_calls`, then why `not tool_calls and should_terminate_sequence`?